diesel/src/query_builder/debug.rs

```diff
@@ -0,0 +1,57 @@
+use super::{QueryBuilder, BuildQueryResult};
+use types::NativeSqlType;
+
+/// The structure is used to to return SQL without the need for a connection.
+/// You should not need to use this directly.  Instead use the debug_sql! macro
+/// and the print_sql! macro.
+///
+/// # Example
+///
+/// # Examples
+///
+/// ### Returning SQL from a count statment:
+/// #
+/// # ```rust
+/// # // example requires setup for users table
+/// # use diesel::users::dsl::*;
+/// # use diesel::query_builder::QueryFragment;
+/// # use diesel::query_builder::debug::DebugQueryBuilder;
+/// #
+/// # fn main() {
+/// #     let mut query_builder = DebugQueryBuilder::new();
+/// #     command = users.count();
+/// #     QueryFragment::to_sql(&command, &mut query_builder).unwrap();
+/// #     query_builder.sql  // => "SELECT COUNT(*) FROM users"
+/// # }
+/// # ```
```

Comment:
Probably shouldn't have this as part of the pubic api.  In that case, should this doc section be removed and the `#[doc(hidden)]` be added?